src/lib/components/tabsv2/Tabsv2.component.js

```diff
@@ -0,0 +1,402 @@
+          nextScrollStart += scrollButtonEndRef.current.getBoundingClientRect()
+            .width;
+        }
+        scroll(nextScrollStart, { animation });
+      }
+    },
+    // we do not want to re-trigger this function if display.start / display.end are modified
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [getTabsMeta],
+  );
+
+  const updateScrollButtonState = useCallback(() => {
+    if (tabsRef.current) {
+      const { scrollLeft, scrollWidth, clientWidth } = tabsRef.current;
+      let showStartScroll = scrollLeft > 1;
+      let showEndScroll = scrollLeft < scrollWidth - clientWidth - 1;
+
+      if (
+        showStartScroll !== displayScroll.start ||
+        showEndScroll !== displayScroll.end
+      ) {
+        setDisplayScroll({ start: showStartScroll, end: showEndScroll });
+      }
+    }
+  }, [displayScroll.end, displayScroll.start]);
+
+  useEffect(() => {
+    window.addEventListener('resize', updateScrollButtonState);
+    return () => {
+      window.removeEventListener('resize', updateScrollButtonState);
+    };
+  }, [updateScrollButtonState]);
+
+  const handleTabsScroll = useMemo(() => () => updateScrollButtonState(), [
+    updateScrollButtonState,
+  ]);
+
+  useEffect(() => {
+    updateScrollButtonState();
+  });
+
+  // scroll to selected tab on first render
+  useEffect(() => {
+    scrollSelectedIntoView(true);
+  }, [scrollSelectedIntoView, selectedTabIndex]);
+
+  const handleKeyDown = (event) => {
+    const list = tabsListRef.current;
+    const ownerDocument = (list && list.ownerDocument) || document;
+    const currentFocus = ownerDocument.activeElement;
+    const role = currentFocus.getAttribute('role');
+    if (role !== 'tab') {
+      return;
+    }
+
+    switch (event.key) {
+      case 'ArrowLeft':
+        event.preventDefault();
+        moveFocus(list, currentFocus, previousItem);
+        break;
+      case 'ArrowRight':
+        event.preventDefault();
+        moveFocus(list, currentFocus, nextItem);
+        break;
+      case 'Home':
+        event.preventDefault();
+        moveFocus(list, null, nextItem);
+        break;
+      case 'End':
+        event.preventDefault();
+        moveFocus(list, null, previousItem);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const filteredTabsChildren = React.Children.toArray(children).filter(
+    (child) => React.isValidElement(child) && child.type === Tab,
+  );
+
+  useEffect(() => {
+    let hasSelectedTab = false;
+    filteredTabsChildren.forEach((child, index) => {
+      const isSelected = !!matchPath(location.pathname, {
+        path: child.props.path,
+        exact: child.props.exact,
+        strict: child.props.strict,
+        sensitive: child.props.sensitive,
+      });
+      if (isSelected) {
+        setSelectedTabIndex(index);
+        hasSelectedTab = true;
+      }
+    });
+    if (!hasSelectedTab) setSelectedTabIndex(null);
+  }, [location.pathname, filteredTabsChildren]);
```

Comment:
To me this not a side effect of the component so should not be in an effect. Considering this as an effect means that on first render the wrong tab will be displayed. Whereas I think the selectedTabIndex should be computed thanks to `useMemo` as a derived value of the current location.

Reply:
We need to wait a complete render so that the size of tabs is computed and we can scroll to selected tab. Hence we can keep an effect here